### ISSUE TITLE
GoCardless throws 4xx errors 

### ISSUE THREAD

**Issue by johnnye** [thephpleague/omnipay-gocardless#6]
The GoCardless API uses 4xx errors to provide information about the state of the request. In it's current configuration the GoCardless gateway throws these errors. It makes the resulting code quite messy. 

I suggest using something like the stripe library to stop these exceptions bubbling out. 

Like this from https://github.com/thephpleague/omnipay-stripe/blob/master/src/Message/AbstractRequest.php#L105 : 

``` php
// don't throw exceptions for 4xx errors
        $this->httpClient->getEventDispatcher()->addListener(
            'request.error',
            function ($event) {
                if ($event['response']->isClientError()) {
                    $event->stopPropagation();
                }
            }
        );
```

**Comment by greydnls** [comment]
I'm definitely a +1 on that idea. I think Paypal does the same, actually.

**Comment by johnnye** [comment]
This should be fixed with #7 being merged.
